core/src/main/java/org/ehcache/spi/cache/AbstractValueHolder.java

```diff
@@ -129,9 +133,20 @@ public boolean equals(Object obj) {
   }
 
   @Override
-  public float hitRate(TimeUnit unit) {
-    //XXX
-    return 0.0f;
+  public float hitRate(long now, TimeUnit unit) {
+    final long endTime = TimeUnit.NANOSECONDS.convert(now, TimeUnit.MILLISECONDS);
+    final long startTime = TimeUnit.NANOSECONDS.convert(creationTime, nativeTimeUnit());
+    float duration = (endTime - startTime)/(float)TimeUnit.NANOSECONDS.convert(1, unit);
+    return (hits/duration);
+  }
+
+  @Override
+  public long getHits() {
```

Comment:
I know that @alexsnaps already blurred the lines on this one, but could we stick to one naming convention within this class?  Currently only `getId` and `getHits` have a get prefix.